ARGUS/ARGUS_UNet/utils/ARGUS_visualization.py

```diff
@@ -0,0 +1,206 @@
+
+        metric_file = model_filename_base + "val_dice_" + str(vfold_num) + ".npy"
+        metric_values = np.load(metric_file)
+
+        plt.figure("Train", (12, 6))
+
+        plt.subplot(1, 2, 1)
+        plt.title("Epoch Average Loss")
+        x = [i + 1 for i in range(len(epoch_loss_values))]
+        y = epoch_loss_values
+        plt.xlabel("epoch")
+        plt.plot(x, y)
+        plt.ylim([0.0, 0.8])
+
+        plt.subplot(1, 2, 2)
+        plt.title("Val Mean Dice")
+        x = [2 * (i + 1) for i in range(len(metric_values))]
+        y = metric_values
+        plt.xlabel("epoch")
+        plt.plot(x, y)
+        plt.ylim([0.0, 0.8])
+
+        plt.show()
+
+def view_testing_results_vfold(self, model_type="best", run_id=[0], device_num=0):
+    print("VFOLD =", self.vfold_num, "of", self.num_folds - 1)
+
+    test_outputs = [self.test_vfold(model_type, r, device_num) for r in run_id]
+
+    num_runs = len(run_id)
+
+    artery_prior = 1
+    class_artery = 1
+
+    artery_min_size = 20000
+    artery_max_size = 30000
+
+    with torch.no_grad():
+        for image_num, test_data in enumerate(self.test_loader):
+            fname = os.path.basename(
+                self.test_files[self.vfold_num][image_num]["image"]
+            )
+            print("Image:", fname)
+
+            plt.figure("check", (18, 6))
+            num_subplots = (
+                max(self.net_in_channels, num_runs * self.num_classes) + 3
+            )  # 3 = blank + ensemble(needle) + pred
+            subplot_num = 1
+            for c in range(self.net_in_channels):
+                plt.subplot(2, num_subplots, subplot_num)
+                plt.title(f"image")
+                tmpV = test_data["image"][0, c, :, :]
+                plt.imshow(tmpV, cmap="gray")
+                subplot_num += 1
+            plt.subplot(2, num_subplots, num_subplots)
+            plt.title(f"label")
+            tmpV = test_data["label"][0, 0, :, :]
+            for c in range(self.num_classes):
+                tmpV[0, c] = c
+            plt.imshow(tmpV)
+            subplot_num += 1
+
+            # Indent by one plot
+            subplot_num = num_subplots + 1
+            prob_shape = test_outputs[0].shape
+            prob_total = np.zeros(prob_shape)
+            for run_num in range(num_runs):
+                prob = np.empty(prob_shape)
+                run_output = test_outputs[run_num]
+                for c in range(self.num_classes):
+                    itkProb = itk.GetImageFromArray(run_output[c, :, :])
+                    imMathProb = tube.ImageMath.New(itkProb)
+                    imMathProb.Blur(5)
+                    itkProb = imMathProb.GetOutput()
+                    prob[c] = itk.GetArrayFromImage(itkProb)
+                pmin = prob.min()
+                pmax = prob.max()
+                prange = pmax - pmin
+                prob = (prob - pmin) / prange
+                prob[class_artery] = prob[class_artery] * artery_prior
```

Comment:
From here until and including line 158 will need to go into a loop for all classes > 0.    There are issues with doing this for multiple classes (e.g., the ordering in which priors are determined matters, and adjusting the prior for class i could mess up the object size / priors for all classes < i (whose priors are already defined).  This will require some research.